internal/llm/agent/agent.go

```diff
@@ -387,12 +388,23 @@ func (a *agent) processGeneration(ctx context.Context, sessionID, content string
 	if err != nil {
 		return a.err(fmt.Errorf("failed to list messages: %w", err))
 	}
+
+	// sliding window to limit message history
+	maxMessagesInContext := cfg.Options.MaxMessages
+	if maxMessagesInContext > 0 && len(msgs) > maxMessagesInContext {
+		// Keep the first message (usually system/context) and the last N-1 messages
+		msgs = append(msgs[:1], msgs[len(msgs)-maxMessagesInContext+1:]...)
+	}
+
```

Comment:
wouldn't this mean that the agent will loose context? if so, I think the default should be way higher

Reply:
We can't really do this without a good strategy this is going to mess up the behavior of the agent